### PR TITLE
Create minimal seed SDM

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

File: src/atomist.config.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ const machineOptions: ConfigureOptions = {
         "sdm.docker.hub.registry",
         "sdm.docker.hub.user",
         "sdm.docker.hub.password",
```
Comment (cdupuis) [inline src/atomist.config.ts]: Remove the `sdm.docker` options?

File: src/atomist.config.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ const machineOptions: ConfigureOptions = {
...
+        "sdm.generator.spring.project.owner",
+        "sdm.generator.spring.project.repo",
```
Comment (cdupuis) [inline src/atomist.config.ts]: Rename to `sdm.seed.spring.owner` and `sdm.seed.spring.repo`. Also please make them optional and default to a _good_ seed repo.

File: src/machine/machine.ts [hunk 3]
```diff
@@ -20,31 +20,19 @@ import {
...
-    enableDeploy,
-    executeTag,
     summarizeGoalsInGitHubStatus,
-    TagGoal,
 } from "@atomist/sdm-core";
-import { addDockerfile } from "../commands/addDockerfile";
-import {addK8sSupport} from "./kubeSupport";
-import { addNodeSupport } from "./nodeSupport";
-import { addSpringSupport } from "./springSupport";
+import {addSpringSupport} from "./springSupport";
```
Comment (cdupuis) [inline src/machine/machine.ts]: Formatting. Missing spaces.

File: src/machine/machine.ts [hunk 4]
```diff
@@ -20,31 +20,19 @@ import {
...
 export function machine(
     configuration: SoftwareDeliveryMachineConfiguration,
 ): SoftwareDeliveryMachine {
 
     const sdm = createSoftwareDeliveryMachine({
-            name: "Kubernetes Demo Software Delivery Machine",
-            configuration,
-        });
-    addNodeSupport(sdm);
+        name: "Kubernetes Seed Software Delivery Machine",
```
Comment (cdupuis) [inline src/machine/machine.ts]: Name? Probably not _Kubernetes_ something as you deleted this above.

File: src/machine/springSupport.ts [hunk 5]
```diff
@@ -14,154 +14,177 @@
...
-import { DelimitedWriteProgressLogDecorator } from "@atomist/sdm/api-helper/log/DelimitedWriteProgressLogDecorator";
-import { spawnAndWatch } from "@atomist/sdm/api-helper/misc/spawned";
+import {
+    CommonJavaGeneratorConfig,
+    HasSpringBootApplicationClass,
+    MavenBuilder,
+} from "@atomist/sdm-pack-spring/dist";
+import {executableJarDeployer} from "@atomist/sdm-pack-spring/dist/support/java/deploy/executableJarDeployer";
+import {ListLocalDeploys} from "@atomist/sdm-pack-spring/dist/support/maven/deploy/listLocalDeploys";
+import {SpringBootSuccessPatterns} from "@atomist/sdm-pack-spring/dist/support/spring/deploy/localSpringBootDeployers";
```
Comment (cdupuis) [inline src/machine/springSupport.ts]: Missing spaces. Did you run `npm run lint:fix` locally?

File: src/atomist.config.ts (hunk 2 above)
Comment (lievendoclo) [reply src/atomist.config.ts]: Currently using spring-petclinic, but there's a bug that's breaking the package renaming (usage of the word `package` in a top comment)

**Comment by jessitron** [comment]
Pull request auto merged by Atomist.

* 1 approved review by @lievendoclo, @cdupuis
* 3 successful checks

[atomist:generated] [atomist:merge-on-approve]

File: src/machine/springSupport.ts (hunk 5 above)
Comment (lievendoclo) [reply src/machine/springSupport.ts]: `lint:fix` does not fix the spaces. It's an IDE setting.

**Pull request merged**